packages/zoltan2/src/util/Zoltan2_StridedData.hpp

```diff
@@ -158,6 +158,44 @@ class StridedData {
   }
 };
 
+// Helper function needed for T=scalar_t specialization
+// Separate function needed // because, with T != scalar_t, 
+// "array = vec_" // would not compile; 
+// ArrayRCP does not overload "=" operator for different types.
+// Separate helper function needed (outside StridedData class)
+// because cannot specialize member function of templated class.
+template<typename scalar_t, typename T>
+static void getInputArrayHelper(
+  ArrayRCP<const T> &target,
+  const ArrayRCP<const scalar_t> &src)
+{
+  // Create a copy of desired type T
+  // From logic in getInputArray, we know stride == 1.
+  size_t n = src.size();
+  T *tmp = new T [n];
```

Comment:
You could just wrap `tmp` in an `ArrayRCP` from the beginning, using the `ArrayRCP` constructor that takes a raw pointer and `true` (for "owning")